pkg/scheduler/framework/plugins/interpodaffinity/plugin.go

```diff
@@ -71,3 +78,52 @@ func getArgs(obj runtime.Object) (config.InterPodAffinityArgs, error) {
 	}
 	return *ptr, nil
 }
+
+// Updates Namespaces with the set of namespaces identified by NamespaceSelector.
+// If successful, NamespaceSelector is set to nil.
+func mergeAffinityTermNamespaces(at *framework.AffinityTerm, nsLister listersv1.NamespaceLister) error {
+	ns, err := nsLister.List(at.NamespaceSelector)
```

Comment:
@Huang-Wei @alculquicondor I am afraid this will be problematic for clusters with large number of namespaces. I wonder if we should consider supporting all namespaces via empty namespaceselector, which we can special case and support efficiently (by just checking for matching pod label selector since the namespace will always match).

Reply:
When does it start being a problem?

We already have to iterate through all Nodes (is any of that done serially?). IIRC, there's a limit in the number of namespaces supported.

Reply:
I still need to test this on large number of namespaces to see at what point this becomes problematic, but a k8s cluster could have up to 10k namespaces, and consider a "notIn kube-system" namespaceSelector; in this case we will pay a large penalty here in the Lister (I assume it checks every namespace sequentially against the selector) even if those namespaces are mostly empty of pods.

Reply:
The other option would be to lookup the namespace of each pod we test and match it against the selector. The issue here is that labels may change underneath, and so we may want to include namespace objects as part of the scheduler cache and snapshot.

Reply:
`notIn kube-system` would still be quite common, even if we special case "any namespace"

Reply:
Placing kube-system pods on a separate set of nodes using node affinity would likely the more common pattern.

Reply:
but you still have daemonset pods

Reply:
Although they wouldn't match pod selectors.

Maybe it makes sense, but it needs sign-in from sig-auth.

Reply:
Yes, we need to drop the validation in the API PR. I will send an update to the KEP.